platform/persistence/platform-persistence-commands/src/main/java/org/codice/ddf/persistence/commands/AbstractStoreCommand.java

```diff
@@ -81,4 +85,30 @@ public Object execute() {
 
   /** Calls a command that operates on the Persistent Store service. */
   abstract void storeCommand() throws PersistenceException;
+
+  protected String createCql(String user, String cql) {
+    if (StringUtils.isBlank(user)) return "";
+    if (StringUtils.isNotBlank(cql)) {
+      cql = "( " + cql + ") AND user='" + user + "'";
+    } else {
+      cql = "user='" + user + "'";
+    }
+    return cql;
+  }
+
+  protected List<Map<String, Object>> getResults() throws PersistenceException {
+
+    List<Map<String, Object>> results = new ArrayList<>();
+    List<Map<String, Object>> pagedResults;
+    int startIndex = 0;
+    int pageSize = 1000;
+
+    do {
+      pagedResults = persistentStore.get(type, cql, startIndex, pageSize);
+      results.addAll(pagedResults);
+      startIndex += pageSize;
+    } while (pagedResults.size() > 0);
```

Comment:
❗ How many items do we expect to be in the persistent store? Is there a risk of running out of memory?

Reply:
updated to export per batch